File: Assignments/Assignment4/Programming/BitXor.java
import java.util.Scanner;

public class BitXor {

    public static int bitXor(int x, int y) {
        return ~(~(x & ~y) & ~(~x & y));
    }

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        System.out.print("Enter x: ");
        int x = sc.nextInt();

        System.out.print("Enter y: ");
        int y = sc.nextInt();

        System.out.println("bitXor(" + x + ", " + y + ") = " + bitXor(x, y));

        sc.close();
    }
}

File: Assignments/Assignment4/Programming/BitXor.c
#include <stdio.h>

int bitXor(int x, int y) {
    return ~(~(x & ~y) & ~(~x & y));
}

int main() {

    int x, y;

    printf("Enter x: ");
    scanf("%d", &x);

    printf("Enter y: ");
    scanf("%d", &y);

    int result = bitXor(x, y);

    printf("bitXor(%d, %d) = %d\n", x, y, result);

    return 0;
}
